CommonClient.py

```diff
@@ -459,6 +459,14 @@ async def send_connect(self, **kwargs: typing.Any) -> None:
         await self.send_msgs([payload])
         await self.send_msgs([{"cmd": "Get", "keys": ["_read_race_mode"]}])
 
+    async def check_locations(self, locations: typing.Collection[int]) -> set[int]:
+        """Send new location checks to the server. Returns set of actually new locations that were send."""
+        locations = set(locations)
+        locations.update(self.missing_locations)
```

Comment:
```suggestion
        locations &= self.missing_locations
```
or i guess set.intersection_update() ?
set.update() is for union not intersection